zipline/assets/futures.py

```diff
@@ -126,15 +126,13 @@ def _maybe_update_current_chain(self):
             list
                 The up-to-date current chain, a list of Future objects.
         """
-        dt = self._get_datetime()
-
-        if (self._last_updated is None) or (self._last_updated != dt):
+        if (self._last_updated is None)\
+                or (self._last_updated != self.as_of_date):
```

Comment:
Nope, but there is no need to update unless the as_of_date property has changed, so this would avoid some unnecessary updates.